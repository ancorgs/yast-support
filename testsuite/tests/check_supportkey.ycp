// testedfiles: supportio.ycp

{

    include "testsuite.ycp";

    include "support/supportio.ycp";

    map<string,any> keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "male",
       "company": "M&M",
       "street": "",
       "zip": "",
       "country": "Italy",
       "state": "",
       "city": "",
       "email": "WI@IW.jh",
       "key": "WI-CATCH22-Yo-Yo"
    ];

    DUMP (keydata);
    // keydata ok
	TEST (``(check_supportkey(keydata)), [ $[], $[], $[] ], nil);
    DUMP("----------------------");

    keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "",
       "company": "M&M",
       "street": "",
       "zip": "",
       "country": "Italy",
       "state": "",
       "city": "",
       "email": "wi@iw.jh",
       "key": "WI-CATCH22-Yo-Yo"
    ];

    DUMP (keydata);
    // missing salutation
	TEST (``(check_supportkey(keydata)), [ $[], $[], $[] ], nil);
    DUMP("----------------------");

    keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "Mr",
       "company": "M&M",
       "street": "",
       "zip": "",
       "country": "Italy",
       "state": "",
       "city": "",
       "email": "wi@iw",
       "key": "WI-CATCH22-Yo-Yo"
    ];

    DUMP (keydata);
    // bad email adress
	TEST (``(check_supportkey(keydata)), [ $[], $[], $[] ], nil);
    DUMP("----------------------");

    keydata =
    $[ "name1": "Washington",
       "name2": "Irwing",
       "salutation": "Mr",
       "company": "M&M",
       "street": "",
       "zip": "",
       "country": "Italy",
       "state": "",
       "city": "",
       "email": "wi@iw.jh",
       "key": "WI-CATCH22"
    ];

    DUMP (keydata);
    // short support key
	TEST (``(check_supportkey(keydata)), [ $[], $[], $[] ], nil);
}
