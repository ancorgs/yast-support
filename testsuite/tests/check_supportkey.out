Dump	$["city":"", "company":"M&M", "country":"Italy", "email":"WI@IW.jh", "key":"WI-CATCH22-Yo-Yo", "name1":"Washington", "name2":"Irwing", "salutation":"male", "state":"", "street":"", "zip":""]
Return	
Dump	----------------------
Dump	$["city":"", "company":"M&M", "country":"Italy", "email":"wi@iw.jh", "key":"WI-CATCH22-Yo-Yo", "name1":"Washington", "name2":"Irwing", "salutation":"", "state":"", "street":"", "zip":""]
Return	Check either Mrs. or Mr.
Dump	----------------------
Dump	$["city":"", "company":"M&M", "country":"Italy", "email":"wi@iw", "key":"WI-CATCH22-Yo-Yo", "name1":"Washington", "name2":"Irwing", "salutation":"Mr", "state":"", "street":"", "zip":""]
Return	Enter a valid e-mail address
Dump	----------------------
Dump	$["city":"", "company":"M&M", "country":"Italy", "email":"wi@iw.jh", "key":"WI-CATCH22", "name1":"Washington", "name2":"Irwing", "salutation":"Mr", "state":"", "street":"", "zip":""]
Return	Please enter your support key
